{
  "topic_metadata": {
    "id": 4406,
    "title": "WebApp Git error: \"Failed to push changes: There is a mismatch between file name and format\"",
    "slug": "webapp-git-error-failed-to-push-changes-there-is-a-mismatch-between-file-name-and-format",
    "posts_count": 8,
    "created_at": "2025-06-01T06:06:25.504Z",
    "last_posted_at": "2025-06-04T09:57:52.389Z",
    "views": 73,
    "like_count": 8,
    "reply_count": 4,
    "has_accepted_answer": true,
    "accepted_answer_post_number": 5,
    "accepted_answer_username": "laurmaedje",
    "tags": [],
    "category_id": 5,
    "participant_count": 4,
    "word_count": 358
  },
  "posts": [
    {
      "id": 13515,
      "post_number": 1,
      "username": "David",
      "name": "David",
      "created_at": "2025-06-01T06:06:25.566Z",
      "updated_at": "2025-06-01T06:06:25.566Z",
      "cooked": "<p>I am experiencing this (closed issue):</p><aside class=\"onebox githubissue\" data-onebox-src=\"https://github.com/typst/webapp-issues/issues/244\">\n  <header class=\"source\">\n\n      <a href=\"https://github.com/typst/webapp-issues/issues/244\" target=\"_blank\" rel=\"noopener nofollow ugc\">github.com/typst/webapp-issues</a>\n  </header>\n\n  <article class=\"onebox-body\">\n    <div class=\"github-row\">\n  <div class=\"github-icon-container\" title=\"Issue\" data-github-private-repo=\"false\">\n\t  <svg width=\"60\" height=\"60\" class=\"github-icon\" viewBox=\"0 0 14 16\" aria-hidden=\"true\"><path fill-rule=\"evenodd\" d=\"M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z\"></path></svg>\n  </div>\n\n  <div class=\"github-info-container\">\n    <h4>\n      <a href=\"https://github.com/typst/webapp-issues/issues/244\" target=\"_blank\" rel=\"noopener nofollow ugc\">Cannot move/rename matlab source files: \"There is a mismatch between file name and format\"</a>\n    </h4>\n\n    <div class=\"github-info\">\n      <div class=\"date\">\n        opened <span class=\"discourse-local-date\" data-format=\"ll\" data-date=\"2023-12-22\" data-time=\"10:13:23\" data-timezone=\"UTC\">10:13AM - 22 Dec 23 UTC</span>\n      </div>\n\n        <div class=\"date\">\n          closed <span class=\"discourse-local-date\" data-format=\"ll\" data-date=\"2024-11-14\" data-time=\"18:07:12\" data-timezone=\"UTC\">06:07PM - 14 Nov 24 UTC</span>\n        </div>\n\n      <div class=\"user\">\n        <a href=\"https://github.com/ToxicMushroom\" target=\"_blank\" rel=\"noopener nofollow ugc\">\n          <img alt=\"\" src=\"https://forum.typst.app/uploads/default/original/2X/7/7fdcc30a6e41f90541cf97958afb9af80534738f.png\" class=\"onebox-avatar-inline\" width=\"20\" height=\"20\" data-dominant-color=\"9A9162\">\n          ToxicMushroom\n        </a>\n      </div>\n    </div>\n\n    <div class=\"labels\">\n        <span style=\"display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;\">\n          bug\n        </span>\n    </div>\n  </div>\n</div>\n\n  <div class=\"github-row\">\n    <p class=\"github-body-container\">### Description\n\nThis is probably caused by some more general issue.\nIn the lin<span class=\"show-more-container\"><a href=\"\" rel=\"noopener\" class=\"show-more\">…</a></span><span class=\"excerpt hidden\">ked project you cannot rename `hello` to `hello.m`\nAnd you cannot move `hello.m` into the `folder` (Also breaks when trying to move a matlab file out of a folder)\n\n\n### Reproduction URL\n\nhttps://typst.app/project/wgVrz1lPR9SUi1MGz6buGq\n\n### Browsers\n\nFirefox\n\n### OS\n\nLinux</span></p>\n  </div>\n\n  </article>\n\n  <div class=\"onebox-metadata\">\n    \n    \n  </div>\n\n  <div style=\"clear: both\"></div>\n</aside>\n<aside class=\"onebox githubissue\" data-onebox-src=\"https://github.com/typst/webapp-issues/issues/471\">\n  <header class=\"source\">\n\n      <a href=\"https://github.com/typst/webapp-issues/issues/471\" target=\"_blank\" rel=\"noopener nofollow ugc\">github.com/typst/webapp-issues</a>\n  </header>\n\n  <article class=\"onebox-body\">\n    <div class=\"github-row\">\n  <div class=\"github-icon-container\" title=\"Issue\" data-github-private-repo=\"false\">\n\t  <svg width=\"60\" height=\"60\" class=\"github-icon\" viewBox=\"0 0 14 16\" aria-hidden=\"true\"><path fill-rule=\"evenodd\" d=\"M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z\"></path></svg>\n  </div>\n\n  <div class=\"github-info-container\">\n    <h4>\n      <a href=\"https://github.com/typst/webapp-issues/issues/471\" target=\"_blank\" rel=\"noopener nofollow ugc\">\"There is a mismatch between file name and format\" error when pushing to git repo</a>\n    </h4>\n\n    <div class=\"github-info\">\n      <div class=\"date\">\n        opened <span class=\"discourse-local-date\" data-format=\"ll\" data-date=\"2024-10-23\" data-time=\"04:19:03\" data-timezone=\"UTC\">04:19AM - 23 Oct 24 UTC</span>\n      </div>\n\n        <div class=\"date\">\n          closed <span class=\"discourse-local-date\" data-format=\"ll\" data-date=\"2024-11-15\" data-time=\"16:23:55\" data-timezone=\"UTC\">04:23PM - 15 Nov 24 UTC</span>\n        </div>\n\n      <div class=\"user\">\n        <a href=\"https://github.com/peng1999\" target=\"_blank\" rel=\"noopener nofollow ugc\">\n          <img alt=\"\" src=\"https://forum.typst.app/uploads/default/original/2X/e/eee49a925be7726fc382e8b7a3457776510ab9e4.png\" class=\"onebox-avatar-inline\" width=\"20\" height=\"20\" data-dominant-color=\"63A3EB\">\n          peng1999\n        </a>\n      </div>\n    </div>\n\n    <div class=\"labels\">\n        <span style=\"display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;\">\n          bug\n        </span>\n    </div>\n  </div>\n</div>\n\n  <div class=\"github-row\">\n    <p class=\"github-body-container\">### Description\n\nAfter writing the commit message and click \"push\" button, the e<span class=\"show-more-container\"><a href=\"\" rel=\"noopener\" class=\"show-more\">…</a></span><span class=\"excerpt hidden\">rror occurs and refuse to proceed pushing.\n\nThis is the layout of the repo:\n&lt;img width=\"360\" alt=\"image\" src=\"https://github.com/user-attachments/assets/b915b2db-bdb2-4d20-bfca-494f7aec8535\"&gt;\n\nThe `latex` folder is not used but I want to keep that. Is that cause a problem?\n\n### Reproduction URL\n\n_No response_\n\n### Browsers\n\nFirefox\n\n### OS\n\nmacOS</span></p>\n  </div>\n\n  </article>\n\n  <div class=\"onebox-metadata\">\n    \n    \n  </div>\n\n  <div style=\"clear: both\"></div>\n</aside>\n\n<p>I am not quite sure what to do. I only have .typ, .png, .jpg, .jpeg, .bib and .svg files in my project. Everything worked fine the day before yesterday, but I can neither push or pull using Git.</p>\n<p>This is in the WebApp.<br>\n<div class=\"lightbox-wrapper\"><a class=\"lightbox\" href=\"https://forum.typst.app/uploads/default/original/2X/5/52ccfbdc7541f3c0228411d2f15e87f45b1bba00.png\" data-download-href=\"https://forum.typst.app/uploads/default/52ccfbdc7541f3c0228411d2f15e87f45b1bba00\" title=\"Screenshot 2025-05-30 160439\"><img src=\"https://forum.typst.app/uploads/default/optimized/2X/5/52ccfbdc7541f3c0228411d2f15e87f45b1bba00_2_143x500.png\" alt=\"Screenshot 2025-05-30 160439\" data-base62-sha1=\"bOukmpYRk3IsIdh9r5wcshmnq4o\" width=\"143\" height=\"500\" srcset=\"https://forum.typst.app/uploads/default/optimized/2X/5/52ccfbdc7541f3c0228411d2f15e87f45b1bba00_2_143x500.png, https://forum.typst.app/uploads/default/optimized/2X/5/52ccfbdc7541f3c0228411d2f15e87f45b1bba00_2_214x750.png 1.5x, https://forum.typst.app/uploads/default/optimized/2X/5/52ccfbdc7541f3c0228411d2f15e87f45b1bba00_2_286x1000.png 2x\" data-dominant-color=\"212027\"><div class=\"meta\"><svg class=\"fa d-icon d-icon-far-image svg-icon\" aria-hidden=\"true\"><use href=\"#far-image\"></use></svg><span class=\"filename\">Screenshot 2025-05-30 160439</span><span class=\"informations\">376×1307 74.4 KB</span><svg class=\"fa d-icon d-icon-discourse-expand svg-icon\" aria-hidden=\"true\"><use href=\"#discourse-expand\"></use></svg></div></a></div><br>\n<img src=\"https://forum.typst.app/uploads/default/original/2X/0/0176eba9ae8c54ae0da0a3fa0d70af3e1f0a6c54.png\" alt=\"Screenshot 2025-05-30 173102\" data-base62-sha1=\"cXglf2f6Q4npmMRXxXEvAc8Lo8\" width=\"530\" height=\"50\"></p>",
      "raw": "I am experiencing this (closed issue):\nhttps://github.com/typst/webapp-issues/issues/244\nhttps://github.com/typst/webapp-issues/issues/471\n\nI am not quite sure what to do. I only have .typ, .png, .jpg, .jpeg, .bib and .svg files in my project. Everything worked fine the day before yesterday, but I can neither push or pull using Git.\n\nThis is in the WebApp.\n![Screenshot 2025-05-30 160439|143x500](upload://bOukmpYRk3IsIdh9r5wcshmnq4o.png)\n![Screenshot 2025-05-30 173102|530x50](upload://cXglf2f6Q4npmMRXxXEvAc8Lo8.png)",
      "reply_to_post_number": null,
      "reply_count": 0,
      "quote_count": 0,
      "like_count": 1,
      "reads": 29,
      "score": 65.8,
      "accepted_answer": false,
      "trust_level": 0,
      "user_id": 1681
    },
    {
      "id": 13518,
      "post_number": 2,
      "username": "Konstantin",
      "name": "Konstantin",
      "created_at": "2025-06-01T10:55:37.832Z",
      "updated_at": "2025-06-01T11:01:21.910Z",
      "cooked": "<p>I’m facing the exact same issue. I disconnected the repo, deleted the github repo and made a new one and reconnected which worked. Edited only one file and as I wanted to push or pull I got the same error message as you. This is very unfortunate as I just bought the pro plan to have the online and offline sync ability for my (soon due) thesis.</p>",
      "raw": "I'm facing the exact same issue. I disconnected the repo, deleted the github repo and made a new one and reconnected which worked. Edited only one file and as I wanted to push or pull I got the same error message as you. This is very unfortunate as I just bought the pro plan to have the online and offline sync ability for my (soon due) thesis.",
      "reply_to_post_number": null,
      "reply_count": 1,
      "quote_count": 0,
      "like_count": 1,
      "reads": 24,
      "score": 24.8,
      "accepted_answer": false,
      "trust_level": 0,
      "user_id": 1683
    },
    {
      "id": 13520,
      "post_number": 3,
      "username": "David",
      "name": "David",
      "created_at": "2025-06-01T11:03:37.584Z",
      "updated_at": "2025-06-01T11:36:45.268Z",
      "cooked": "<p>I am sorry to hear that!<br>\nI have not yet found a solution, and am still experiencing the problem.<br>\n(I am also in the exact same boat, having bought the pro plan to sync my bachelor’s thesis)</p>",
      "raw": "I am sorry to hear that!\nI have not yet found a solution, and am still experiencing the problem.\n(I am also in the exact same boat, having bought the pro plan to sync my bachelor's thesis)",
      "reply_to_post_number": 2,
      "reply_count": 0,
      "quote_count": 0,
      "like_count": 1,
      "reads": 24,
      "score": 34.8,
      "accepted_answer": false,
      "trust_level": 0,
      "user_id": 1681
    },
    {
      "id": 13557,
      "post_number": 4,
      "username": "AMendoza",
      "name": "AMendoza",
      "created_at": "2025-06-02T01:58:44.646Z",
      "updated_at": "2025-06-02T01:58:44.646Z",
      "cooked": "<p>I am experiencing this error as well.</p>",
      "raw": "I am experiencing this error as well.",
      "reply_to_post_number": null,
      "reply_count": 0,
      "quote_count": 0,
      "like_count": 1,
      "reads": 23,
      "score": 24.6,
      "accepted_answer": false,
      "trust_level": 1,
      "user_id": 1691
    },
    {
      "id": 13571,
      "post_number": 5,
      "username": "laurmaedje",
      "name": "Laurenz Mädje",
      "created_at": "2025-06-02T10:47:05.124Z",
      "updated_at": "2025-06-02T10:47:05.124Z",
      "cooked": "<p>The problem should be fixed now. It was introduced on Friday in the course of fixing another bug. We apologize for the inconvenience.</p>",
      "raw": "The problem should be fixed now. It was introduced on Friday in the course of fixing another bug. We apologize for the inconvenience.",
      "reply_to_post_number": null,
      "reply_count": 2,
      "quote_count": 0,
      "like_count": 3,
      "reads": 21,
      "score": 59.2,
      "accepted_answer": true,
      "trust_level": 4,
      "user_id": 2
    },
    {
      "id": 13574,
      "post_number": 6,
      "username": "David",
      "name": "David",
      "created_at": "2025-06-02T11:04:44.324Z",
      "updated_at": "2025-06-02T11:04:44.324Z",
      "cooked": "<p>Thank you. It seems fixed.</p>\n<p>In the meantime, I’ve had to manually push my changes to GitHub (downloading the files). However, when I now tried to pull the changes from the WebApp, it presented me with some, what I guess are, merge conflicts. Even though I chose to keep all the Typst files (from the WebApp) multiple changes were made to my files, that looked a lot like some kind of corruption. Luckily, I had a backup and have restored it. If you are interested in the “corruption”: <a href=\"https://github.com/ETIMC/Report/commit/9009678c76e9137a592bf87857291c910cf1b312\" rel=\"noopener nofollow ugc\">https://github.com/ETIMC/Report/commit/9009678c76e9137a592bf87857291c910cf1b312</a></p>",
      "raw": "Thank you. It seems fixed.\n\nIn the meantime, I've had to manually push my changes to GitHub (downloading the files). However, when I now tried to pull the changes from the WebApp, it presented me with some, what I guess are, merge conflicts. Even though I chose to keep all the Typst files (from the WebApp) multiple changes were made to my files, that looked a lot like some kind of corruption. Luckily, I had a backup and have restored it. If you are interested in the \"corruption\": [https://github.com/ETIMC/Report/commit/9009678c76e9137a592bf87857291c910cf1b312](https://github.com/ETIMC/Report/commit/9009678c76e9137a592bf87857291c910cf1b312)",
      "reply_to_post_number": 5,
      "reply_count": 1,
      "quote_count": 0,
      "like_count": 0,
      "reads": 21,
      "score": 19.0,
      "accepted_answer": false,
      "trust_level": 0,
      "user_id": 1681
    },
    {
      "id": 13584,
      "post_number": 7,
      "username": "laurmaedje",
      "name": "Laurenz Mädje",
      "created_at": "2025-06-02T12:45:33.561Z",
      "updated_at": "2025-06-02T12:45:33.561Z",
      "cooked": "<p>Thanks for the details and the backup branch, it will be useful in investigating that issue.</p>",
      "raw": "Thanks for the details and the backup branch, it will be useful in investigating that issue.",
      "reply_to_post_number": 6,
      "reply_count": 0,
      "quote_count": 0,
      "like_count": 1,
      "reads": 20,
      "score": 18.8,
      "accepted_answer": false,
      "trust_level": 4,
      "user_id": 2
    },
    {
      "id": 13673,
      "post_number": 8,
      "username": "Konstantin",
      "name": "Konstantin",
      "created_at": "2025-06-04T09:57:52.389Z",
      "updated_at": "2025-06-04T09:57:52.389Z",
      "cooked": "<p>Thank you for fixing and the update! Works like a charm again now. Absolutely back in love with Typst &lt;3</p>",
      "raw": "Thank you for fixing and the update! Works like a charm again now. Absolutely back in love with Typst <3",
      "reply_to_post_number": 5,
      "reply_count": 0,
      "quote_count": 0,
      "like_count": 0,
      "reads": 12,
      "score": 2.2,
      "accepted_answer": false,
      "trust_level": 0,
      "user_id": 1683
    }
  ],
  "scraped_at": "2025-06-10T09:46:57.535963"
}